### ISSUE TITLE
Formatter: not separate def with visibility modifier

### ISSUE THREAD

**Issue by MakeNowJust** [crystal-lang/crystal#4133]
This script:

```crystal
def foo
end
def bar
end
```

is formatted to:

```crystal
def foo
end

def bar
end
```

Even if it is, this script is not formatted:

```crystal
def foo
end
private def bar
end
```

I think it is a bug.

- - -

```console
$ crystal version
Crystal 0.21.1 (2017-03-07) LLVM 3.9.1
```

**Comment by asterite** [comment]
@MakeNowJust It formats correctly for me, and the spec you added in #4134 is already passing for me. Are we doing something different?

**Comment by MakeNowJust** [comment]
@asterite Sorry, it's just my mistake. Correct case is here:

```crystal
private def foo
end
private def bar
end
```

This case is not formatted by current `crystal tool format`. It is fixed by #4134, so I'll fix spec in this.

**Comment by asterite** [comment]
@MakeNowJust ありがとう！
